src/common/stream_coder.h

```diff
@@ -0,0 +1,23 @@
+#ifndef STREAM_CODER_H
+#define STREAM_CODER_H
+
+#include "allocator.h"
+#include <nan.h>
+
+using namespace v8;
+
+class StreamCoder : public Nan::ObjectWrap {
+  public:
+    const uint8_t* next_in;
+    size_t available_in;
```

Comment:
Done… I don’t know how long the streaming compression API has been around in brotli, but do you think it might make sense to use that for compressing, too? It probably would reduce the code differences between decoding and encoding quite a bit more.